controllers/etcdpeer_controller.go

```diff
@@ -16,20 +20,106 @@ type EtcdPeerReconciler struct {
 	Log logr.Logger
 }
 
+const (
+	EtcdImage = "quay.io/coreos/etcd:v3.2.27"
+)
+
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=replicaset,verbs=get;update;patch;create
+
+func defineReplicaSet(etcdPeer etcdv1alpha1.EtcdPeer) (appsv1.ReplicaSet, error) {
+	var replicas int32 = 1
+
+	return appsv1.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:          make(map[string]string),
+			Annotations:     make(map[string]string),
+			Name:            etcdPeer.Name,
+			Namespace:       etcdPeer.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(&etcdPeer, etcdv1alpha1.GroupVersion.WithKind("EtcdPeer"))},
+		},
+		Spec: appsv1.ReplicaSetSpec{
+			// This will *always* be 1. Other peers are handled by other EtcdPeers.
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "etcd",
+					// Using the EtcdPeer's name as a label limits what the name can be
+					"peer": etcdPeer.Name,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "etcd",
+						// Using the EtcdPeer's name as a label limits what the name can be
+						"peer": etcdPeer.Name,
+					},
+					Annotations: make(map[string]string),
+					Name:        etcdPeer.Name,
+					Namespace:   etcdPeer.Namespace,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "etcd",
+							Image: EtcdImage,
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
 
 func (r *EtcdPeerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("etcdpeer", req.NamespacedName)
+	ctx := context.Background()
```

Comment:
do you think it would be sensible to have a timeout on this?

Reply:
Perhaps, although given the super early state I'm unsure of what would be a sensible timeout value. If we begin to make live etcd client operations, we may have to wait a few seconds...

Reply:
we won't remember to change this :D  can we set to ~10 seconds and increase it later if needed?

Reply:
> we won't remember to change this :D

It's a fair point. 😂  I've used `context.WithTimeout` to set a 10 second timeout on the context.